packages/flutter/test/material/tab_bar_theme_test.dart

```diff
@@ -1791,4 +1791,60 @@ void main() {
     targetRect = const Rect.fromLTRB(275.0, 0.0, 325.0, 48.0);
     expectIndicatorAttrs(tabBarBox, rect: rect, targetRect: targetRect);
   });
+
+  testWidgets('splashBorderRadius is passed from TabBarThemeData', (WidgetTester tester) async {
+    const Color hoverColor = Color(0xfff44336);
+    const double radius = 20;
+    await tester.pumpWidget(
+      Material(
+        child: Localizations(
+          locale: const Locale('en', 'US'),
+          delegates: const <LocalizationsDelegate<dynamic>>[
+            DefaultMaterialLocalizations.delegate,
+            DefaultWidgetsLocalizations.delegate,
+          ],
+          child: DefaultTabController(
+            length: 1,
+            child: TabBarTheme(
+              data: TabBarThemeData(
+                 splashBorderRadius: BorderRadius.circular(radius),
+              ),
+              child: TabBar(
+                overlayColor: MaterialStateProperty.resolveWith<Color>(
+                  (Set<MaterialState> states) {
+                    if (states.contains(MaterialState.hovered)) {
+                      return hoverColor;
+                    }
+                    return Colors.black54;
+                  },
+                ),
+
+                tabs: const <Widget>[
+                  Tab(
+                    child: Text(''),
+                  ),
+                ],
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+    await tester.pumpAndSettle();
+    final TestGesture gesture = await tester.createGesture(kind: PointerDeviceKind.mouse, pointer: 1);
+    await gesture.moveTo(tester.getCenter(find.byType(Tab)));
+    await tester.pumpAndSettle();
+    final RenderObject object = tester.allRenderObjects.firstWhere((RenderObject element) => element.runtimeType.toString() == '_RenderInkFeatures');
+    expect(
+      object,
+      paints..rrect(
+        color: hoverColor,
+        rrect: RRect.fromRectAndRadius(
+          tester.getRect(find.byType(InkWell)),
+          const Radius.circular(radius)
+        ),
+      ),
+    );
```

Comment:
Could we get rid of the `element.runtimeType.toString() == '_RenderInkFeatures'` here?

```suggestion
    final Finder findInkWell = find.byType(InkWell);
    final BuildContext context = tester.element(findInkWell);
    expect(
      Material.of(context),
      paints..rrect(
        color: hoverColor,
        rrect: RRect.fromRectAndRadius(
          tester.getRect(findInkWell),
          const Radius.circular(radius),
        ),
      ),
    );
```